### PR TITLE
TASK-923

### PR DESCRIPTION
**Pull request by juanrizetta**
TASK-923

### CODE REVIEW

**Comment by juanrizetta** [comment]
Task linked: [TASK-923 RGA solr hosts not configured in k8s](https://app.clickup.com/t/36631768/TASK-923)

File: opencga-app/app/cloud/docker/opencga-init/override_yaml.py [hunk 1]
```diff
@@ -74,6 +75,13 @@
         storage_config["clinical"]["hosts"].clear()
     storage_config["clinical"]["hosts"].insert(i, clinical_host.strip())
 
+# Inject RGA hosts
+rga_hosts = args.rga_hosts.replace('\"','').replace('[','').replace(']','').split(",")
```
Comment (j-coll) [inline opencga-app/app/cloud/docker/opencga-init/override_yaml.py:79]: It'd have been nice to extract this into a method, as this same code is triplicated now.

File: opencga-app/app/cloud/docker/opencga-init/override_yaml.py [hunk 2]
```diff
@@ -9,6 +9,7 @@
 parser.add_argument("--database-prefix", required=False)
 parser.add_argument("--search-hosts", required=True)
 parser.add_argument("--clinical-hosts", required=True)
+parser.add_argument("--rga-hosts", required=True)
```
Comment (j-coll) [inline opencga-app/app/cloud/docker/opencga-init/override_yaml.py:12]: This "required=True" will break the `test_override_yaml.py` . You'll need to check and fix the test.

**Review by j-coll — CHANGES_REQUESTED** [review]

File: opencga-app/app/cloud/docker/opencga-init/override_yaml.py (hunk 2 above)
Comment (juanrizetta) [reply opencga-app/app/cloud/docker/opencga-init/override_yaml.py]: I supposed same value of required than clinical and search hosts.
I change to requierd=False

File: opencga-app/app/cloud/docker/opencga-init/override_yaml.py (hunk 1 above)
Comment (juanrizetta) [reply opencga-app/app/cloud/docker/opencga-init/override_yaml.py]: I have removed the code to a method, I have adjusted it to reuse it in 5 places

File: opencga-app/app/cloud/docker/opencga-init/override_yaml.py [hunk 3]
```diff
@@ -9,6 +9,7 @@
 parser.add_argument("--database-prefix", required=False)
 parser.add_argument("--search-hosts", required=True)
 parser.add_argument("--clinical-hosts", required=True)
+parser.add_argument("--rga-hosts", required=False)
```
Comment (j-coll) [inline opencga-app/app/cloud/docker/opencga-init/override_yaml.py:12]: It'd be nice to add a test in test_override_yaml.py to test this change

File: opencga-app/app/cloud/docker/opencga-init/override_yaml.py [hunk 4]
```diff
@@ -56,23 +57,22 @@
...
-
-# Inject clinical hosts
-clinical_hosts = args.clinical_hosts.replace('\"','').replace('[','').replace(']','').split(",")
-for i, clinical_host in enumerate(clinical_hosts):
-    if i == 0:
-        # If we are overriding the default hosts,
-        # clear them only on the first iteration
-        storage_config["clinical"]["hosts"].clear()
-    storage_config["clinical"]["hosts"].insert(i, clinical_host.strip())
+def hostOverride(hosts_var,server,component):
```
[the remark this answers is not in the document]
Comment (juanrizetta) [reply opencga-app/app/cloud/docker/opencga-init/override_yaml.py:60]: yes, i realized it :).
is fixed in last commit

**Review by j-coll — APPROVED** [review]

**Pull request merged**